### DISCUSSION TITLE
Cannot we use something like Solidjs instead of Preact

### DISCUSSION THREAD

**Discussion by kawpii** [freshframework/fresh#237] [Ideas]
Solidjs : https://solidjs.com/

**Comment by lucacasonato** [comment]
Not right now, no.

**Reply by verlor** [reply]
are there  any plans on the near future to support other/s frameworks ?

**Reply by lucacasonato** [reply]
If the given framework provide the same pluggability as Preact (https://preactjs.com/guide/v10/options)  which is required to make islands work, I can consider it.

**Reply by shishkin** [reply]
What is the dealbreaker in pluggability? Astro [supports Solid](https://github.com/withastro/astro/tree/main/packages/integrations/solid). How is their islands implementation different?

**Reply by marvinhagemeister** [reply]
They have written a custom compiler that converts `.astro` templates into the required formats. Afaik they don't allow you to mix and match server rendered content and client side one like we do.

**Reply by adamzerner** [reply]
I suspect that supporting other frameworks would add a good deal of complexity and decrease velocity a good amount. Astro is the only metaframework I know of that supports multiple frameworks. Personally, I prefer and suspect it'd be best for the Fresh team to [say no](https://basecamp.com/gettingreal/05.3-start-with-no) here.

**Comment by andykais** [comment]
If we ever get to a place which supports multiple jsx compilers, these discussions over in the solidjs repo may be helpful:
- https://github.com/solidjs/solid/discussions/873
- https://github.com/solidjs/solid/discussions/332

**Comment by robvanvolt** [comment]
What about nano-jsx? Is this a potential replacement for preact?
https://nanojsx.io/

**Comment by saolof** [comment]
What concrete advantages does Solidjs offer over Preact? The latter is much more mature, and while solidjs is faster than react, the perf advantage over preact seems to be smaller than the standard deviation in measurement results according to this benchmark with solid having a worse worst-case performance, while also requiring a more complex toolchain:
https://webcomponents.dev/blog/all-the-ways-to-make-a-web-component/

The entire point of Deno is to require zero build tools (not even a typescript compiler since the runtime will just accept .ts files and jsx), which fresh also aims for, so moving to a framework which requires a compiler feels like a step backwards in that sense. You should just be able to copy the raw source files from the git repo to the runtime container.

**Reply by shishkin** [reply]
I'm not sure if web components context is different, but in [this benchmark](https://krausest.github.io/js-framework-benchmark/current.html) Solid shows pretty consistent and significant performance gain over Preact.

Besides performance, Solid might be better positioned for Fresh with its island architecture since Solid is aiming at [eliminating need for component hydration](https://dev.to/this-is-learning/patterns-for-building-javascript-websites-in-2022-5a93).

**Reply by saolof** [reply]
Ah, the performance metrics are different and consistent between the two. Preact wins by a small margin on startup time and time to interactive, while solid does wins on RAM usage and operations on collections of ~1000 elements, while both contribute negligibly to bundle size.

By eliminating hydration, do you mean having streaming SSR?

**Reply by shishkin** [reply]
> By eliminating hydration, do you mean having streaming SSR?

@ryansolid describes it like this in https://dev.to/this-is-learning/resumability-wtf-2gcm:

> This is known as Partial Hydration; you may have seen a version of this technique, [Islands](https://jasonformat.com/islands-architecture/), in frameworks like [Marko](https://markojs.com/), [Astro](https://astro.build/), or [Fresh](https://fresh.deno.dev/). The difference is resumable frameworks can do this at a sub-component level, much smaller than Islands, and they can do it automatically.
